consensus/polybft/consensus_runtime.go

```diff
@@ -0,0 +1,851 @@
+	isEndOfEpoch := c.isEndOfEpoch(pendingBlockNumber)
+
+	ff := &fsm{
+		config:         c.config.PolyBFTConfig,
+		parent:         parent,
+		backend:        c.config.blockchain,
+		polybftBackend: c.config.polybftBackend,
+		blockBuilder:   blockBuilder,
+		validators:     newValidatorSet(types.BytesToAddress(parent.Miner), epoch.Validators), // TODO: Nemanja - check this
+		isEndOfEpoch:   isEndOfEpoch,
+		isEndOfSprint:  isEndOfSprint,
+		epoch:          epoch.Number,
+		logger:         c.logger.Named("fsm"),
+	}
+
+	var systemState SystemState
+
+	var nextStateSyncExecutionIdx uint64
+
+	if c.IsBridgeEnabled() {
+		systemState, err = c.getSystemState(c.lastBuiltBlock)
+		if err != nil {
+			return nil, err
+		}
+
+		nextStateSyncExecutionIdx, err = systemState.GetNextExecutionIndex()
+		if err != nil {
+			return nil, err
+		}
+
+		ff.stateSyncExecutionIndex = nextStateSyncExecutionIdx
+	}
+
+	ff.postInsertHook = func() error {
+		if c.IsBridgeEnabled() {
+			if isEndOfEpoch && ff.commitmentToSaveOnRegister != nil {
+				if err := c.state.insertCommitmentMessage(ff.commitmentToSaveOnRegister); err != nil {
+					return err
+				}
+
+				if err := c.buildBundles(
+					epoch, ff.commitmentToSaveOnRegister.Message, nextStateSyncExecutionIdx); err != nil {
+					return err
+				}
+			}
+		}
+
+		c.NotifyProposalInserted(ff.block)
+
+		return nil
+	}
+
+	if c.IsBridgeEnabled() {
+		nextRegisteredCommitmentIndex, err := systemState.GetNextCommittedIndex()
+		if err != nil {
+			return nil, err
+		}
+
+		if isEndOfEpoch {
+			commitment, err := c.getCommitmentToRegister(epoch, nextRegisteredCommitmentIndex)
+			if err != nil {
+				if errors.Is(err, ErrCommitmentNotBuilt) {
+					c.logger.Info("[FSM] Have no built commitment to register",
+						"epoch", epoch.Number, "from state sync index", nextRegisteredCommitmentIndex)
+				} else if errors.Is(err, errQuorumNotReached) {
+					c.logger.Info("[FSM] Not enough votes to register commitment",
+						"epoch", epoch.Number, "from state sync index", nextRegisteredCommitmentIndex)
+				} else {
+					return nil, err
+				}
+			}
+
+			ff.proposerCommitmentToRegister = commitment
+		}
+
+		if isEndOfSprint {
+			if err != nil {
+				return nil, err
+			}
+
+			if err := c.state.cleanCommitments(nextStateSyncExecutionIdx); err != nil {
+				return nil, err
+			}
+
+			nonExecutedCommitments, err := c.state.getNonExecutedCommitments(nextStateSyncExecutionIdx)
+			if err != nil {
+				return nil, err
+			}
+
+			if len(nonExecutedCommitments) > 0 {
+				bundlesToExecute, err := c.state.getBundles(nextStateSyncExecutionIdx, maxBundlesPerSprint)
+				if err != nil {
+					return nil, err
+				}
+
+				ff.commitmentsToVerifyBundles = nonExecutedCommitments
+				ff.bundleProofs = bundlesToExecute
+			}
+		}
+	}
+
+	// TODO: Nemanja
+	// if isEndOfEpoch {
+	// 	uptimeCounter, err := c.calculateUptime(parent)
+	// 	if err != nil {
+	// 		return nil, err
+	// 	}
+
+	// 	ff.uptimeCounter = uptimeCounter
+	// }
```

Comment:
Leftover?

Reply:
Will be changed in later PR-s. This is just one of many PR-s necessary to move polybft to edge. We can not move everything at once